packages/ethpm-v3/lib/ethpm-v3.js

```diff
@@ -0,0 +1,485 @@
+    let ethpm;
+    try {
+      ethpm = await EthPM.configure({
+        manifests: "ethpm/manifests/v3",
+        storage: "ethpm/storage/ipfs",
+        registries: "ethpm/registries/web3"
+      }).connect({
+        provider: registryProvider,
+        registryAddress: options.ethpm.registry.address,
+        ipfs: {
+          host: options.ethpm.ipfsHost,
+          port: options.ethpm.ipfsPort,
+          protocol: options.ethpm.ipfsProtocol
+        }
+      });
+    } catch (err) {
+      throw new TruffleError(`Unable to configure ethpm: ${err.message}`);
+    }
+
+    options.logger.log("Finding publishable artifacts...");
+    let artifacts;
+    try {
+      artifacts = await utils.getPublishableArtifacts(options, ethpm);
+    } catch (err) {
+      throw new TruffleError(
+        `Unable to collect artifacts: ${err}. Try compiling your contracts.`
+      );
+    }
+
+    // Fetch ethpm.json config
+    let ethpmConfig;
+    try {
+      ethpmConfig = JSON.parse(
+        fs.readFileSync(
+          path.join(options.working_directory, "ethpm.json"),
+          "utf8"
+        )
+      );
+      if (
+        typeof ethpmConfig.name === "undefined" ||
+        typeof ethpmConfig.version === "undefined"
+      ) {
+        throw new TruffleError(
+          "Invalid ethpm.json: Must contain a 'name' and 'version'."
+        );
+      }
+    } catch (err) {
+      throw new TruffleError("Invalid ethpm.json configuration detected.");
+    }
+
+    // Includes all installed ethpm packages in published package
+    const buildDependencies = utils.fetchInstalledBuildDependencies(
+      options.working_directory
+    );
+
+    const ethpmFields = {
+      sources: artifacts.resolvedSources,
+      contractTypes: artifacts.resolvedContractTypes,
+      compilers: artifacts.resolvedCompilers,
+      deployments: artifacts.resolvedDeployments,
+      buildDependencies: buildDependencies
+    };
+
+    options.logger.log(`Generating package manifest...`);
+    const targetConfig = Object.assign(ethpmFields, ethpmConfig);
+    let pkg;
+    let manifest;
+    let manifestUri;
+    try {
+      pkg = await ethpm.manifests.read(JSON.stringify(targetConfig));
+      manifest = await ethpm.manifests.write(pkg);
+      manifestUri = await ethpm.storage.write(manifest);
+    } catch (err) {
+      throw new TruffleError(
+        `Error generating manifest from contract artifacts: ${err}`
+      );
+    }
+
+    options.logger.log(`Publishing package to registry...`);
+    const w3 = new Web3(registryProvider);
+    const fromAddress = await w3.eth.getCoinbase();
+    const encodedTxData = ethpm.registries.registry.methods
+      .release(ethpmConfig.name, ethpmConfig.version, manifestUri.href)
+      .encodeABI();
+    const tx = await w3.eth.signTransaction({
+      from: fromAddress,
+      to: options.ethpm.registry.address,
+      gas: 4712388,
+      gasPrice: 100000000000,
+      data: encodedTxData
+    });
+    try {
```

Comment:
It seems like there might be a better mechanism available somewhere in truffle that manages sending txs with a more user-friendly interface? Could you point me in the right direction...